### ISSUE TITLE
Cant UPDATE ArrayField to empty array

### ISSUE THREAD

**Issue by zahlio** [yandex/ClickHouse#3381]
I have the following table:

```
create table grank.keyword
(
	id UInt64 default CAST(0, 'UInt64'),
	domain_id UInt64 default CAST(0, 'UInt64'),
	tags Array(String),
)
engine = MergeTree() PARTITION BY floor(domain_id, -3) ORDER BY id SETTINGS index_granularity = 8192;
```

And run the following query:
```
ALTER TABLE keyword UPDATE
tags=[],
WHERE id=2555 AND domain_id=9;
```
And get the following error:
```
infi.clickhouse_orm.database.ServerError: Illegal columns Const(Array(Nothing)) and Array(String) of second (then) and third (else) arguments of function if (44)
```

**Comment by filimonov** [comment]
Try

```
ALTER TABLE keyword UPDATE
tags=emptyArrayString(),
WHERE id=2555 AND domain_id=9;
```

**Comment by zahlio** [comment]
@filimonov thanks that worked for me

**Comment by ztlpn** [comment]
That was fixed in #4243 (available in 19.1.7 and 19.3.3)
